make/helm.mk

```diff
@@ -0,0 +1,68 @@
+##@ Helm Charts
+
+.PHONY: helm-build
+helm-build: $(KUSTOMIZE) $(OPERATOR_SDK) $(YQ) manifests ## Build the helm chart from kustomize manifests
+	# Set desired operator image and related limitador image
+	V="$(RELATED_IMAGE_LIMITADOR)" $(YQ) eval '(select(.kind == "Deployment").spec.template.spec.containers[].env[] | select(.name == "RELATED_IMAGE_LIMITADOR").value) = strenv(V)' -i config/manager/manager.yaml
+	# Replace the controller image
+	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
+	# Build the helm chart templates from kustomize manifests
+	$(KUSTOMIZE) build config/helm > charts/limitador-operator/templates/manifests.yaml
+	V="$(VERSION)" $(YQ) eval '.version = strenv(V)' -i charts/limitador-operator/Chart.yaml
+
+.PHONY: helm-install
+helm-install: $(HELM) ## Install the helm chart
+	# Install the helm chart in the cluster
+	$(HELM) install limitador-operator charts/limitador-operator
+
+.PHONY: helm-uninstall
+helm-uninstall: $(HELM) ## Uninstall the helm chart
+	# Uninstall the helm chart from the cluster
+	$(HELM) uninstall limitador-operator
+
+.PHONY: helm-upgrade
+helm-upgrade: $(HELM) ## Upgrade the helm chart
+	# Upgrade the helm chart in the cluster
+	$(HELM) upgrade limitador-operator charts/limitador-operator
+
+.PHONY: helm-package
+helm-package: $(HELM) ## Package the helm chart
+	# Package the helm chart
+	$(HELM) package charts/limitador-operator
+
+# GitHub Token with permissions to upload to the release assets
+GITHUB_TOKEN ?= <YOUR-TOKEN>
+# GitHub Release ID, to find the release_id query the GET /repos/{owner}/{repo}/releases/latest or GET /repos/{owner}/{repo}/releases endpoints
+RELEASE_ID ?= <RELEASE-ID>
+# GitHub Release Asset ID, it can be find in the output of the uploaded asset
+ASSET_ID ?= <ASSET-ID>
+# GitHub Release Asset Browser Download URL, it can be find in the output of the uploaded asset
+BROWSER_DOWNLOAD_URL ?= <BROWSER-DOWNLOAD-URL>
+# Github repo name for the helm charts repository
+HELM_REPO_NAME ?= helm-charts
+ifeq (0.0.0,$(VERSION))
+CHART_VERSION = $(VERSION)-dev
+else
+CHART_VERSION = $(VERSION)
+endif
+
+.PHONY: helm-upload-package
+helm-upload-package: $(HELM) ## Upload the helm chart package to the GitHub release assets
```

Comment:
have you considered something like https://github.com/marketplace/actions/upload-files-to-a-github-release ??

Reply:
Thanks for sharing, I didn't consider using any github action for this, maybe it's cleaner that way. I'll give it a try! 👍🏼